pkg/datapath/linux/probes/probes.go

```diff
@@ -240,6 +240,50 @@ func (p *ProbeManager) SystemConfigProbes() error {
 	return nil
 }
 
+// ExitKernelCheck exits execution of cilium CLI kernel-check early in case the config file
+// is missing (or running on a kernel before 3.18)
+func (p *ProbeManager) ExitKernelCheck() error {
```

Comment:
Function looks good but name and description are misleading. The function does not stop execution (it would if we had an `os.Exit(1)`, but I think it's better to just return an error the way you did). It simply checks (indirectly) if we could find the config file. Maybe something like `ConfigFound()` would be a better fit.

But actually I don't know if we should perform this check at all in this PR, as #10317 should introduce a more robust check by validating the presence of the config file just the way bpftool does. My opinion is that we should use that check once its available (can be a follow-up if this PR is merged first), which would mean removing this function here entirely.

Reply:
@qmonnet thanks, addressed your comments. 

As for `ExitKernelCheck`: as of now the function is still in place. As discussed, we wanted to use the function as introduced by #10317 - but this PR has not yet merged into the mainline. 

I can remove the function with this PR and after the above mentioned PR has been merged I'd rebase against master und use that very function. 
Alternatively we could merge this one first (not including `ExitKernelCheck`) and as a follow-up I'd  do a follow-up-PR incorporating `KernelConfigAvailable()`. Either way I'm fine.

Reply:
> Alternatively we could merge this one first (not including `ExitKernelCheck`) and as a follow-up I'd do a follow-up-PR incorporating `KernelConfigAvailable()`.

I think we can go for that. The other PR looks ready and should be merged soon anyway.